### PR TITLE
flambda2/Cmm changes for float16 bigarrays

### PR DESCRIPTION
**Pull request by mshinwell**
As subject.  Plus one tiny fix to allow flambda2 to compile.

### CODE REVIEW

File: backend/cmm_helpers.ml [hunk 1]
```diff
@@ -1743,6 +1744,7 @@ let bigarray_elt_size_in_bytes : Lambda.bigarray_kind -> int = function
 
 let bigarray_word_kind : Lambda.bigarray_kind -> memory_chunk = function
   | Pbigarray_unknown -> assert false
+  | Pbigarray_float16 -> Sixteen_unsigned
```
Comment (Gbury) [inline backend/cmm_helpers.ml:1747]: We need a new `memory_chunk` constructor to ensure that this is loaded as a `float` and not an unsigned 16-bit integer

**Review by Gbury — CHANGES_REQUESTED** [review]

File: backend/cmm_helpers.ml (hunk 1 above)
Comment (lthls) [reply backend/cmm_helpers.ml:1747]: I don't think this is the right solution. If we want to stay as close as possible to upstream here, we should keep the memory as a 16-bit integer but call the appropriate functions (`float_of_float16` and `float16_of_float` from `Cmm_helpers`) to convert between the naked float and the 16-bit integer around loading and storing.

Comment (Gbury) [reply backend/cmm_helpers.ml:1747]: That seems like the good solution indeed

Comment (mshinwell) [reply backend/cmm_helpers.ml:1779]: Fixed

**Review by Gbury — APPROVED** [review]

**Pull request merged**